### PR TITLE
Filesys name

### PR DESCRIPTION
**Pull request by yihanzhen**
Allows an empty path "" returned and to represent the root directory when calling fileSystem.getPath(""), so that methods in java.nio.file.Files (e.g., Files.list()) can function correctly for root directory.

### CODE REVIEW

File: google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java [hunk 1]
```diff
@@ -384,6 +385,34 @@ public void testDeleteRecursive() throws IOException {
...
+    // test absolute path
+    Path rootPath = fs.getPath("");
+    List<String> objectNames = new ArrayList<String>();
+    for (Path path : Files.newDirectoryStream(rootPath)) {
+      objectNames.add(path.toString());
+    }
+    Collections.sort(objectNames);
+    assertThat(objectNames.size()).isEqualTo(2);
+    assertThat(objectNames.get(0)).isEqualTo(BIG_FILE);
+    assertThat(objectNames.get(1)).isEqualTo(SML_FILE);
```
Comment (pongad) [inline google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java]: You don't need to sort. Try `assertThat(objectNames).containsExactly(BIG_FILE, SML_FILE)`

File: google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java [hunk 2]
```diff
@@ -384,6 +385,34 @@ public void testDeleteRecursive() throws IOException {
...
+    
+    // test relative path
+    rootPath = fs.getPath(".");
+    for (Path path : Files.newDirectoryStream(rootPath)) {
+      objectNames.add(path.toString());
+    }
+    Collections.sort(objectNames);
+    assertThat(objectNames.size()).isEqualTo(2);
+    assertThat(objectNames.get(0)).isEqualTo(BIG_FILE);
+    assertThat(objectNames.get(1)).isEqualTo(SML_FILE);
```
Comment (pongad) [inline google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java]: Same as above.

File: google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStoragePath.java [hunk 3]
```diff
@@ -74,6 +74,8 @@ public String bucket() {
    * Returns path converted to a {@link BlobId} so I/O can be performed.
    */
   BlobId getBlobId() {
+    checkArgument(!path.toString().isEmpty(), 
+      "I/O not allowed on empty Google Cloud Storage object names.");
```
Comment (pongad) [inline google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStoragePath.java]: I think this error message is awfully abstruse. Maybe just "object names cannot be empty"?

File: google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java [hunk 4]
```diff
@@ -397,20 +397,15 @@ public void testListFilesInRootDirectory() throws IOException {
     for (Path path : Files.newDirectoryStream(rootPath)) {
       objectNames.add(path.toString());
     }
-    Collections.sort(objectNames);
-    assertThat(objectNames.size()).isEqualTo(2);
-    assertThat(objectNames.get(0)).isEqualTo(BIG_FILE);
-    assertThat(objectNames.get(1)).isEqualTo(SML_FILE);
+    assertThat(objectNames.size()).containsExactly(BIG_FILE, SML_FILE);
```
Comment (pongad) [inline google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java]: Just `objectNames` right?

**Comment by hzyi-google** [comment]
They should be good now

**Comment by pongad** [comment]
LGTM @garrettjonesgoogle do you also want to take a look?

**Comment by garrettjonesgoogle** [comment]
I have no problem with the code change. Could you update the commit message so that it states what you are doing in this change? ("filesys name" isn't really meaningful)

**Comment by pongad** [comment]
Actually, please hold off on this for a bit. I have a concern with the way UNIX file names like "." and ".." are treated. I'll follow up in an email.

**Comment by hzyi-google** [comment]
@garrettjonesgoogle Thanks for the remind. Updated.

**Pull request merged**